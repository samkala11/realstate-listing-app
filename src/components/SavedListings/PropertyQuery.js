import React, {useEffect, useState} from 'react';
import Property from '../Property/Property';
import Paper from '@material-ui/core/Paper';
import Skeleton from '@material-ui/lab/Skeleton';


import { getListing } from '../../utils/listings_api_util';

const PropertyQuery = ({mlsId, docId, deleteListing}) => {

    const [listingInfo, setListingInfo] = useState(null);

    window.listingInfoStateQuery = listingInfo;

    useEffect(() => {
        getListing(mlsId)
        .then(res => {
            console.log('async get 1 listing');
            console.log(res);
            setListingInfo(res);
        })

    }, [mlsId])
    

    return(
        <>

            {listingInfo ? 
            <Paper data-testid="property-query-container" id="property-container" elevation={3}>
                <Property
                    key = { listingInfo.mlsId }
                    mlsId = {listingInfo.mlsId } 
                    photosArray = {listingInfo ? listingInfo.photos : []}
                    price = {listingInfo.listPrice}
                    bedrooms = { listingInfo && listingInfo.property ? listingInfo.property.bedrooms : null}
                    bathsFull = { listingInfo && listingInfo.property ? listingInfo.property.bathsFull : null}
                    bathsHalf = {  listingInfo && listingInfo.property ? listingInfo.property.bathsHalf : null}
                    stories = {listingInfo && listingInfo.property ? listingInfo.property.stories : null}
                    listingDate = {listingInfo.listDate}
                    saveProperty = {deleteListing}
                    saved ={true}
                    id ={docId}
                    
                />  
            </Paper> :
            <div data-testid="property-query-container">   
                <Skeleton variant="rect" width={350} height={280} />
                <Skeleton variant="text" />
                <Skeleton variant="text" />
                <Skeleton variant="text" />
            </div>
}


        </>
    )
}

export default PropertyQuery;