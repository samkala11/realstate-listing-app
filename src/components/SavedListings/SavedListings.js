import React, {useState, useEffect} from 'react';
import 'firebase/firestore';
import PropertyQuery from './PropertyQuery';

const SavedListings = ({fireStore}) => {
    let savedPropertiesRef;
    var currentUserPropertiesQuery;
    if (fireStore) {

        savedPropertiesRef = fireStore.collection('savedProperties');
        currentUserPropertiesQuery = savedPropertiesRef.where("userId", "==", localStorage.getItem('userId'));
    }
    const [savedProperties, setSavedProperties] = useState([]);
    window.savedPropertiesState = savedProperties;

    useEffect(() => {
        getSavedProperties();
    }, [])

    const deleteSavedListing = (a, b, id) => {
        savedPropertiesRef.doc(id).delete().then(() => {getSavedProperties()})    
    }

    const getSavedProperties = () => {

        currentUserPropertiesQuery.get().then( snapshot => {
            const savedPropertiesArr = [];
            snapshot.docs.forEach( (doc, ind) => {
                savedPropertiesArr.push(doc.data())
                console.log('adding saved properties in Array')
                savedPropertiesArr[ind]['id'] = doc.id;
                console.log(doc.id)
            })
            setSavedProperties(savedPropertiesArr)
        })
    }


    return (
        <div className="saved-listings">
            <div className="saved-listing-container">
            
            { savedProperties && savedProperties.map(savedProperty => (
                <div key={savedProperty.mlsId}> 
                    {/* <Button  id="delete-button" onClick={() => {
                        deleteSavedListing(savedProperty.id)
                    }}> x </Button> */}
                    <PropertyQuery 
                        mlsId={savedProperty.mlsId}
                        docId = {savedProperty.id}
                        deleteListing = {deleteSavedListing}
                    />

                </div>
            ))}
            </div>

        </div>
    ) 
}

export default SavedListings;