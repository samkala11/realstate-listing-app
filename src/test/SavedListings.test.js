// import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SavedListings from '../components/SavedListings/SavedListings';

test('matches SavedListing snapshot with given props', () => {
    const tree = renderer.create(<SavedListings 
        />).toJSON();
    expect(tree).toMatchSnapshot();
})