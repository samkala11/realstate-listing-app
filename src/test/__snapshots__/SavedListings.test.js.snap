// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches SavedListing snapshot with given props 1`] = `
<div
  className="saved-listings"
>
  <div
    className="saved-listing-container"
  />
</div>
`;
